assets/js/modules/analytics-4/datastore/settings.js

```diff
@@ -139,60 +140,112 @@ export async function submitChanges( { select, dispatch } ) {
 
 		// Only make the API request to enable the Enhanced Measurement setting, not to disable it.
 		if ( isEnhancedMeasurementEnabled ) {
-			await dispatch(
-				MODULES_ANALYTICS_4
-			).setEnhancedMeasurementStreamEnabled(
+			const { error } = await updateEnhancedMeasurementSettings( {
+				select,
+				dispatch,
 				propertyID,
 				webDataStreamID,
-				isEnhancedMeasurementEnabled
-			);
+				isEnhancedMeasurementEnabled,
+			} );
 
-			if (
-				select(
-					MODULES_ANALYTICS_4
-				).haveEnhancedMeasurementSettingsChanged(
-					propertyID,
-					webDataStreamID
-				)
-			) {
-				const { error } = await dispatch(
-					MODULES_ANALYTICS_4
-				).updateEnhancedMeasurementSettings(
-					propertyID,
-					webDataStreamID
-				);
-
-				if ( error ) {
-					return { error };
-				}
+			if ( error ) {
+				return { error };
+			}
+		}
+	}
+
+	const { error } = await saveSettings( select, dispatch );
+
+	if ( error ) {
+		return error;
+	}
+
+	await API.invalidateCache( 'modules', 'analytics-4' );
+
+	return {};
+}
+
+const saveSettings = async ( select, dispatch ) => {
```

Comment:
Are there any reasons to use an anonymous function style for this top-level function? If not, let's then use the standard declaration. The same applies for other top-level functions written in the arrow function style.